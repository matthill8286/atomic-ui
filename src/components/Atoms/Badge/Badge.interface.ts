export type BadgeActionType = 'prominent' | 'secondary' | 'inverted' | 'disabled'

export interface BadgeType {
  id?: number | string
  name: string
  actionType?: BadgeActionType
}

export interface BadgeProps {
  actionType?: BadgeActionType
  isFirst?: boolean
  isBadgeLine?: boolean
  badges: (BadgeType | null)[]
  className?: string
}

export interface StyledBadgeProps {
  actionType?: BadgeActionType
  isFirst?: boolean
  isBadgeLine?: boolean
  className?: string
}
