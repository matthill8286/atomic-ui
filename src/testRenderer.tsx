import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { saiyanTheme } from '@/styles/sc-vars-saiyan'
import { Theme } from '@/types'

export function renderWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>,
  theme = saiyanTheme
) {
  return renderer.create(<ThemeProvider theme={theme}>{tree}</ThemeProvider>).toJSON()
}

export function renderWithThemeAndRouter(tree: React.ReactElement, theme: Theme = saiyanTheme) {
  return renderer
    .create(
      <ThemeProvider theme={theme}>
        <Router>{tree}</Router>
      </ThemeProvider>
    )
    .toJSON()
}

export function renderWithRouter(tree: React.ReactElement) {
  return renderer.create(<Router>{tree}</Router>).toJSON()
}

export function mountWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(<ThemeProvider theme={saiyanTheme}>{tree}</ThemeProvider>)
}

export function mountWithThemeAndRouter<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(
    <ThemeProvider theme={saiyanTheme}>
      <Router>{tree}</Router>
    </ThemeProvider>
  )
}
