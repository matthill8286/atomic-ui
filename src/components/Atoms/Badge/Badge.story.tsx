// @ts-nocheck
import { number, radios } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import { styled } from '@/styles/styled'
import { StorybookWrapper } from '@/utils/StorybookWrapper'
import { Badge } from './Badge'
import { BadgeActionType, BadgeProps } from './Badge.interface'
const actionTypes: { [key: string]: BadgeActionType } = {
  Prominent: 'prominent',
  Secondary: 'secondary',
  Disabled: 'disabled',
}

const badges = [
  {
    id: 0,
    name: 'Featured',
  },
  {
    id: 1,
    name: 'Structured',
  },
  {
    id: 2,
    name: 'Sponsored',
  },
  {
    id: 3,
    name: 'Hot Hot Hot',
  },
  {
    id: 4,
    name: 'Last one',
    actionType: 'secondary',
  },
]

export const StyledLineMaxWidthContainer = styled.div`
  background: lightgrey;
  padding-top: 8px;
  padding-bottom: 8px;
`

const knobs = (): BadgeProps => {
  return {
    actionType: radios('Type', actionTypes, actionTypes.Prominent),
  }
}

storiesOf('Design System/Atoms/Badge', module)
  .add('Default', () => {
    return (
      <StorybookWrapper>
        <Badge badges={badges.slice(0, 1)} {...knobs()} />
      </StorybookWrapper>
    )
  })
  .add('with two badges', () => {
    return (
      <StorybookWrapper>
        <Badge badges={badges.slice(0, 2)} {...knobs()} />
      </StorybookWrapper>
    )
  })
  .add('with two action types', () => {
    return (
      <StorybookWrapper>
        <Badge badges={badges.slice(3, 5)} {...knobs()} />
      </StorybookWrapper>
    )
  })
  .add('BadgeLine', () => {
    const maxWidth = number('Maximum width of container', 600)

    const knobs = (): BadgeProps => {
      return {
        actionType: radios('Type', actionTypes, actionTypes.Prominent),
      }
    }
    return (
      <StorybookWrapper>
        <StyledLineMaxWidthContainer style={{ maxWidth }}>
          <Badge badges={badges} isBadgeLine {...knobs()} />
        </StyledLineMaxWidthContainer>
      </StorybookWrapper>
    )
  })
