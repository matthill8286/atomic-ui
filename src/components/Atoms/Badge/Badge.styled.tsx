import { math } from 'polished'
import { css, styled, ThemedStyledProps } from '@/styles/styled'
import { Theme } from '@/types/theme'
import { Typo } from '../Typography'
import { StyledBadgeProps } from './Badge.interface'

type ThemedBadgeProps = ThemedStyledProps<StyledBadgeProps, Theme>

const skewFactor = 10
const tailWidth = 10

const resolveColor = ({ actionType, theme }: ThemedBadgeProps) =>
  actionType === 'disabled' ? theme.color.black : theme.color.primary

export const skewLeft = ({ actionType, theme }: ThemedBadgeProps) => css`
  &:before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: ${tailWidth + 2}px;
    height: ${math(theme.font.size.xl + '+ 2px')}; /* theme font xl plus the border */
    bottom: 0;
    border-radius: ${theme.dimension.borderRadius1};
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
    transform: skewX(-${skewFactor}deg);
    transform-origin: bottom left;

    ${actionType === 'prominent'
      ? css`
          background-color: ${theme.color.primary};
          border: 1px solid ${theme.color.primary};
        `
      : css`
          background-color: ${theme.color.white};
          border-left: 1px solid ${resolveColor};
          border-top: 1px solid ${resolveColor};
          border-bottom: 1px solid ${resolveColor};
        `}
  }
`
export const skewRight = ({ actionType, theme }: ThemedBadgeProps) => css`
  &:after {
    content: '';
    position: absolute;
    top: 0;
    right: 2px;
    width: ${tailWidth + 2}px;
    height: ${math(theme.font.size.xl + '+ 2px')}; /* theme font xl plus the border */
    bottom: 0;
    border-radius: ${theme.dimension.borderRadius1};
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
    transform: skewX(-${skewFactor}deg);
    border: 1px solid ${resolveColor};

    ${actionType === 'prominent'
      ? css`
          background-color: ${theme.color.primary};
        `
      : css`
          background-color: ${theme.color.white};
          border-left: none;
        `}
  }
`

export const StyledTypo = styled(Typo)<StyledBadgeProps>(
  ({ actionType, theme }) => css`
    ${actionType === 'prominent'
      ? css`
          background-color: ${theme.color.primary};
          color: ${theme.color.white};
        `
      : css`
          background-color: ${theme.color.white};
          color: ${resolveColor};
        `}
    ${actionType === 'inverted' &&
      css`
        background-color: ${theme.color.grey6};
        color: ${theme.color.white};
      `}
    text-align: center;
    display: flex;
    align-items: center;
    padding-left: ${theme.spacing.base.md};
    padding-right: ${theme.spacing.base.md};
    margin-left: 0;
    margin-right: ${tailWidth - 1}px;
    box-sizing: content-box;
    height: ${math(theme.font.size.xl + ' + 2px')}; /* theme font xl plus the border */
    overflow: auto;
  `
)

export const StyledBadgeLineWrapper = styled.div<StyledBadgeProps>`
  ${({ isBadgeLine, theme }) =>
    isBadgeLine
      ? css`
          display: flex;
          flex-direction: row;
          flex-wrap: wrap;
          overflow: hidden;
          height: ${math(theme.font.size.xl + ' + 4px')};
        `
      : `display: block`};
`

export const StyledBadgeWrapper = styled.div<StyledBadgeProps>(
  ({ theme }) => css`
    display: inline-block;
    position: relative;
    margin-right: ${theme.spacing.base.xs};
    max-width: 100%;
  `
)

export const StyledOverflow = styled.div`
  display: inline-block;
  text-overflow: ellipsis;
  overflow: hidden;
`
