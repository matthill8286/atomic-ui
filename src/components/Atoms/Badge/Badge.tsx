import * as React from 'react'
import { BadgeProps } from './Badge.interface'
import {
  StyledBadgeLineWrapper,
  StyledBadgeWrapper,
  StyledOverflow,
  StyledTypo,
} from './Badge.styled'

export const Badge: React.FC<BadgeProps> = ({
  actionType = 'inverted',
  badges,
  isBadgeLine,
  className,
}) => {
  return (
    <StyledBadgeLineWrapper isBadgeLine className={className}>
      {badges &&
        badges.map((badge, i) => {
          if (!badge) {
            return null
          }
          return (
            <StyledBadgeWrapper
              key={`badge-${badge.name}-${i}`}
              actionType={badge.actionType || actionType}
              isFirst={i === 0}
              isBadgeLine={isBadgeLine}>
              <StyledTypo
                weight="regular"
                fontSize={'xxs'}
                margin="0"
                actionType={badge.actionType || actionType}>
                <StyledOverflow>{badge.name}</StyledOverflow>
              </StyledTypo>
            </StyledBadgeWrapper>
          )
        })}
    </StyledBadgeLineWrapper>
  )
}
