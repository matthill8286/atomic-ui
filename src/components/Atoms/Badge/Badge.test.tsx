import React from 'react'
import { Badge } from '@/components/Atoms/Badge/Badge'
import { renderWithTheme } from '@/testRenderer'

const mockedBadges = [
  {
    id: 0,
    name: 'Badge No.1',
  },
  {
    id: 1,
    name: 'Some other label',
  },
  {
    id: 2,
    name: 'Free shipping',
  },
  {
    id: 3,
    name: 'Hot Hot Hot',
  },
  {
    id: 4,
    name: 'Last one',
  },
]

describe('Badge', () => {
  it('should render', () => {
    const tree = renderWithTheme(<Badge badges={mockedBadges} />)
    expect(tree).toMatchSnapshot()
  })

  it('should render with primary color', () => {
    const tree = renderWithTheme(<Badge badges={mockedBadges} actionType="prominent" />)
    expect(tree).toMatchSnapshot()
  })

  it('should render with secondary color', () => {
    const tree = renderWithTheme(<Badge badges={mockedBadges} actionType="secondary" />)
    expect(tree).toMatchSnapshot()
  })

  it('should render a badge line', () => {
    const tree = renderWithTheme(<Badge isBadgeLine badges={mockedBadges} />)
    expect(tree).toMatchSnapshot()
  })

  it('should render a badge line secondary', () => {
    const tree = renderWithTheme(<Badge badges={mockedBadges} isBadgeLine actionType="secondary" />)
    expect(tree).toMatchSnapshot()
  })

  it('should render a badge line disabled', () => {
    const tree = renderWithTheme(<Badge badges={mockedBadges} isBadgeLine actionType="disabled" />)
    expect(tree).toMatchSnapshot()
  })
})
