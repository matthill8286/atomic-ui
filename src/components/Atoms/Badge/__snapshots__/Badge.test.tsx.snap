// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render 1`] = `
.c2 {
  margin: 0;
  font-size: 12px;
  line-height: 16px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c3 {
  background-color: #FFFFFF;
  color: #00C8DC;
  background-color: #333333;
  color: #FFFFFF;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 24px;
  padding-right: 24px;
  margin-left: 0;
  margin-right: 9px;
  box-sizing: content-box;
  height: 28px;
  overflow: auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  overflow: hidden;
  height: 30px;
}

.c1 {
  display: inline-block;
  position: relative;
  margin-right: 8px;
  max-width: 100%;
}

.c4 {
  display: inline-block;
  text-overflow: ellipsis;
  overflow: hidden;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Badge No.1
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Some other label
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Free shipping
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Hot Hot Hot
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Last one
      </div>
    </span>
  </div>
</div>
`;

exports[`Badge should render a badge line 1`] = `
.c2 {
  margin: 0;
  font-size: 12px;
  line-height: 16px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c3 {
  background-color: #FFFFFF;
  color: #00C8DC;
  background-color: #333333;
  color: #FFFFFF;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 24px;
  padding-right: 24px;
  margin-left: 0;
  margin-right: 9px;
  box-sizing: content-box;
  height: 28px;
  overflow: auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  overflow: hidden;
  height: 30px;
}

.c1 {
  display: inline-block;
  position: relative;
  margin-right: 8px;
  max-width: 100%;
}

.c4 {
  display: inline-block;
  text-overflow: ellipsis;
  overflow: hidden;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Badge No.1
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Some other label
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Free shipping
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Hot Hot Hot
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Last one
      </div>
    </span>
  </div>
</div>
`;

exports[`Badge should render a badge line disabled 1`] = `
.c2 {
  margin: 0;
  font-size: 12px;
  line-height: 16px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c3 {
  background-color: #FFFFFF;
  color: #000000;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 24px;
  padding-right: 24px;
  margin-left: 0;
  margin-right: 9px;
  box-sizing: content-box;
  height: 28px;
  overflow: auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  overflow: hidden;
  height: 30px;
}

.c1 {
  display: inline-block;
  position: relative;
  margin-right: 8px;
  max-width: 100%;
}

.c4 {
  display: inline-block;
  text-overflow: ellipsis;
  overflow: hidden;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Badge No.1
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Some other label
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Free shipping
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Hot Hot Hot
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Last one
      </div>
    </span>
  </div>
</div>
`;

exports[`Badge should render a badge line secondary 1`] = `
.c2 {
  margin: 0;
  font-size: 12px;
  line-height: 16px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c3 {
  background-color: #FFFFFF;
  color: #00C8DC;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 24px;
  padding-right: 24px;
  margin-left: 0;
  margin-right: 9px;
  box-sizing: content-box;
  height: 28px;
  overflow: auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  overflow: hidden;
  height: 30px;
}

.c1 {
  display: inline-block;
  position: relative;
  margin-right: 8px;
  max-width: 100%;
}

.c4 {
  display: inline-block;
  text-overflow: ellipsis;
  overflow: hidden;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Badge No.1
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Some other label
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Free shipping
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Hot Hot Hot
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Last one
      </div>
    </span>
  </div>
</div>
`;

exports[`Badge should render with primary color 1`] = `
.c2 {
  margin: 0;
  font-size: 12px;
  line-height: 16px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c3 {
  background-color: #00C8DC;
  color: #FFFFFF;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 24px;
  padding-right: 24px;
  margin-left: 0;
  margin-right: 9px;
  box-sizing: content-box;
  height: 28px;
  overflow: auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  overflow: hidden;
  height: 30px;
}

.c1 {
  display: inline-block;
  position: relative;
  margin-right: 8px;
  max-width: 100%;
}

.c4 {
  display: inline-block;
  text-overflow: ellipsis;
  overflow: hidden;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Badge No.1
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Some other label
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Free shipping
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Hot Hot Hot
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Last one
      </div>
    </span>
  </div>
</div>
`;

exports[`Badge should render with secondary color 1`] = `
.c2 {
  margin: 0;
  font-size: 12px;
  line-height: 16px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c3 {
  background-color: #FFFFFF;
  color: #00C8DC;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 24px;
  padding-right: 24px;
  margin-left: 0;
  margin-right: 9px;
  box-sizing: content-box;
  height: 28px;
  overflow: auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  overflow: hidden;
  height: 30px;
}

.c1 {
  display: inline-block;
  position: relative;
  margin-right: 8px;
  max-width: 100%;
}

.c4 {
  display: inline-block;
  text-overflow: ellipsis;
  overflow: hidden;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Badge No.1
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Some other label
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Free shipping
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Hot Hot Hot
      </div>
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
      fontSize="xxs"
    >
      <div
        className="c4"
      >
        Last one
      </div>
    </span>
  </div>
</div>
`;
